app/controllers/vacancies_controller.rb

```diff
@@ -30,7 +30,8 @@ def show
     return redirect_to(job_path(vacancy), status: :moved_permanently) if old_vacancy_path?(vacancy)
 
     @vacancy = VacancyPresenter.new(vacancy)
-    TrackVacancyPageViewJob.perform_later(vacancy.id) unless authenticated?
+
+    VacancyPageView.new(vacancy).track unless authenticated?
```

Comment:
I like this, like you said it's just swapping one Redis call for another makes sense to cut out the middleman